### ISSUE TITLE
Metamask account changed, but the page is not reloaded automatically

### ISSUE THREAD

**Issue by slam9z** [ConsenSys-archive/drizzle-legacy#76]
the account is not polling by the interval as set in the drizzleOption file.

const drizzleOptions = {
    web3: {
        fallback: {
            type: 'ws',
            url: 'wss://ropsten.infura.io/ws'
        }
    },
    contracts: [],
    events: {},
    polls: {
        blocks: 3000,
        accounts: 3000,
    }
}

When I switched the account from Metamask, the page is not reloaded automatically, I have to manually refresh the page. 

Version that I am using:
"drizzle": "^1.1.5",
"drizzle-react": "^1.1.1",
"drizzle-react-components": "^1.1.0",

Please advise!!

**Comment by dangell7** [comment]
I had to implement an auto refresh on another application. Its not web3, that returns a list of accounts. Where meta mask displays all the accounts to you, but designates a chosen account. Something like this will need to be implemented to catch this.

**Comment by JFGHT** [comment]
Indeed. This is also happening to me.

**Comment by eloudsa** [comment]
Hi
I have the same issue with my Dapp. 
If I change the account on Metamask, I detect that the account changed and I will recall my function, but I have the feeling that the data send back by drizzle are not the updated ones.
I have to force a page refresh on Chrome to reload properly the data according the current account selected on metamask.
Did I miss something?
Is that a way to force a refresh on my Drizzle call?
Thanks

**Comment by soundyogi** [comment]
Is this still relevant?
Switching networks reloads my dapp and I did not implement a special reload.

**Comment by DiscRisknBisque** [comment]
@slam9z The polling should be detecting that change. Will try and reproduce.

Added in development tag because an improvement to web3 and accounts logic is coming.

**Comment by samonchain** [comment]
I hit a similar issue (dealing with transaction errors, not page reloads), though I suspect it may be the same underlying problem.

I documented it in my comments of [this open MetaMask issue](https://github.com/MetaMask/metamask-extension/issues/5587) but the basic gist is: in my Drizzle project I can successfully sign a transaction to a public contract function from one account with MetaMask but if I change my account in MetaMask & attempt the same interaction, it fails.

From what I can tell, it appears that Drizzle's [`cacheSend`](https://truffleframework.com/docs/drizzle/getting-started/contract-interaction#-code-cachesend-code-) relies on the MetaMask account that's active on initial page load (when Drizzle is initialized) and does not update if you change MetaMask accounts during the session. I solved this by overriding the `from` address in the options object I passed to `cacheSend` but feel like Drizzle *should* handle this by default.

@DiscRiskandBisque please let me know if you think this is more relevant as a separate issue or feature request. Thanks!

(edited, adding my Drizzle versions):
```
    "drizzle": "^1.2.4",
    "drizzle-react": "^1.2.0",
    "drizzle-react-components": "^1.2.1",
```

**Comment by DiscRisknBisque** [comment]
This is likely two issues in one, closing this to direct discussion to these newer issues:
- #171 Transaction errors not accessible
- #186 `web3` version is incompatible with `wss`
